third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -3688,15 +3815,16 @@ pub mod prop {
     }
 }
 
-use move_core_types::value::{LayoutTag, MoveStruct, MoveValue};
+use crate::delayed_values::delayed_field_id::DelayedFieldID;
+use move_core_types::value::{MoveStruct, MoveValue};
 
 impl ValueImpl {
     pub fn as_move_value(&self, layout: &MoveTypeLayout) -> MoveValue {
         use MoveTypeLayout as L;
 
-        // Make sure to strip all tags from the type layout.
-        if let L::Tagged(LayoutTag::IdentifierMapping(_), layout) = layout {
-            return self.as_move_value(layout.as_ref());
+        // Make sure to strip all kinds from the type layout.
+        if let L::Native(kind, layout) = layout {
+            panic!("impossible to get native layout ({:?}) with {}", kind, layout)
```

Comment:
it doesn't strip the kind now, it actually panics. Should it panic, or return err? previously it just used the layout, in any case code/comment seem clearly off.